### PR TITLE
Workspace dropdown

### LINKED ISSUE #5328: Localhost UI does not show correct component version
## Expected Behavior
When switching component version in localhost UI, Overview and Compositions tabs should show the selected version of the component

## Actual Behavior
When switching between versions of the component, Overview and Compositions tabs always show information for latest version only

## Steps to Reproduce the Problem
1. Create component with two versions
1. In localhost UI switch from latest version to the lower one
1. Observe Overview and Compositions tabs having information for the wrong version (latest instead of whatever is selected)

## Specifications
current (used) bvm version: 0.0.34
latest available bvm version: 0.0.34

current (used) bit version: 0.0.637
latest installed bit version: 0.0.637
latest available bit version: 0.0.637

* Node version: v14.16.0
* npm / yarn version: 7.20.3
* Platform: mac

Additional info: code explorer works fine, cloud version of the UI shows component versions as expected, only localhost version is not working

### PR DESCRIPTION
**Pull request by GiladShoham**
## Proposed Changes

- show the content of previously tagged version on workspace components when choosing another version from the dropdown
- add workspace entry to version drop-down for workspace components
- resolves #5328

### CODE REVIEW

**Review by guysaar223 — APPROVED** [review]

File: scopes/component/component/ui/menu/menu.tsx [hunk 1]
```diff
@@ -73,6 +75,12 @@ function VersionRelatedDropdowns({
       [component.tags]
     ) || [];
 
+  let isWorkspace = false;
+  if (host === 'teambit.workspace/workspace' && versionList.length && versionList[0] !== 'workspace') {
+    versionList.unshift('workspace');
+    isWorkspace = true;
+  }
```
Comment (KutnerUri) [inline scopes/component/component/ui/menu/menu.tsx:82]: don't mutate immutable objects:

```ts
const isWorkspace = host === 'teambit.workspace/workspace';

const versionList =
  useMemo(
    () => {
      const wsLink = [isWorkspace ? 'workspace' : undefined];
      const tagList = component.tags
        ?.toArray()
        .map((tag) => tag?.version?.version)
        .reverse();

      return [...wsLink, ...tagList]
        .filter((x) => x !== undefined)
    },
    [component.tags, isWorkspace]
  );
```

File: scopes/component/ui/version-dropdown/version-dropdown.tsx [hunk 2]
```diff
@@ -29,22 +38,28 @@ export function VersionDropdown({ versions, currentVersion }: VersionDropdownPro
...
+              let to = `?version=${version}`;
+
+              if (version === 'workspace') {
+                to = location.pathname;
+              }
+
               return (
                 <NavLink
-                  to={`?version=${version}`}
+                  to={to}
```
Comment (KutnerUri) [inline scopes/component/ui/version-dropdown/version-dropdown.tsx]:
```
to={version == LOCAL_VERSION  ? '?' : `?version=${version}`}
```

+ extract the text to const:
```ts
const LOCAL_VERSION = "workspace"
```

File: scopes/component/ui/version-dropdown/version-dropdown.tsx [hunk 3]
```diff
@@ -11,16 +11,25 @@ import styles from './version-dropdown.module.scss';
...
       <div className={styles.noVersions}>
         <VersionPlaceholder currentVersion={currentVersion} />
       </div>
     );
   }
+  let currentVersionWithWs = currentVersion;
+
+  if (isWorkspace && !location.search.includes('version')) {
+    currentVersionWithWs = 'workspace';
+  }
```
Comment (KutnerUri) [inline scopes/component/ui/version-dropdown/version-dropdown.tsx:32]: extract this logic to the consumer - menu.tsx

File: scopes/component/component/ui/menu/menu.tsx (hunk 1 above)
Comment (KutnerUri) [reply scopes/component/component/ui/menu/menu.tsx:82]:
```
const currentVersion = search.includes('version') ? component.version : LOCAL_VERSION
```

**Review by KutnerUri — APPROVED** [review]

**Pull request merged**